src/tools/clippy/tests/ui/into_iter_on_ref.fixed

```diff
@@ -11,7 +11,6 @@ fn main() {
 
     let _ = vec![1, 2, 3].into_iter();
     let _ = (&vec![1, 2, 3]).iter(); //~ ERROR: equivalent to `.iter()
-    let _ = vec![1, 2, 3].into_boxed_slice().iter(); //~ ERROR: equivalent to `.iter()
```

Comment:
This test shouldn't be removed. If the lint now no longer triggers for it, the `ERROR` annotation should be removed and perhaps this line moved to the "should not lint" section of the test file, should that exist.

If it still triggers the lint, but shouldn't, than the Clippy lint code has to be fixed. If that should be the case, I can help you with that, if necessary.

Reply:
The lint no longer applies, since it no longer is redundant. 

In its stead, we now issue an edition migration warning BOXED_SLICE_INTO_ITER. I would need to remove the run-rustfix from this file, too, since the warning from this lint produces two suggestions which are mutually incompatible with each other.

Reply:
So the Clippy lint no longer triggers? That's good.

Thanks for the explanation. I should've read more of the PR changes to find out myself. In that case removing this test case is the best thing to do.